jenkins/Jenkinsfile

```diff
@@ -0,0 +1,109 @@
+def secrets = [
+  [path: 'secret/jenkins/github', engineVersion: 2, secretValues: [
+    [envVar: 'PRIVATE_REPO_GITHUB_TOKEN', vaultKey: 'private_token']]],
+  [path: 'secret/jenkins/npm', engineVersion: 2, secretValues: [
+    [envVar: 'NPM_TOKEN', vaultKey: 'token']]],
+]
+
+def configuration = [vaultUrl: "${env.VAULT_URL}",  vaultCredentialId: 'vault-app-role', engineVersion: 2]
+
+pipeline {
+  agent {
+    kubernetes {
+      label 'common-ui'
+      defaultContainer 'jnlp'
+      yamlFile 'jenkins/build-pod.yaml'
+    }
+  }
+  
+  environment {
+    MAIN_BRANCH = 'master'
+    WORKSPACE = "${env.WORKSPACE}"
+    BRANCH = "${env.BRANCH}"
+    PROJECT = "cloudify-ui-common"
+  }
+
+  options {
+    checkoutToSubdirectory('cloudify-ui-common')
+    buildDiscarder(logRotator(numToKeepStr:'30'))
+    timeout(time: 60, unit: 'MINUTES')
+    timestamps()
+  }
+  stages {
+    stage('Build') {
+      steps {
+        container('node'){
+          dir("${env.WORKSPACE}/${env.PROJECT}") {
+            echo "Install dependencies"
+            sh 'npm ci --prefer-offline'
+            
+            echo "Build application"
+            sh 'npm run build'
+
+            echo "Run static analysis and check code style"
+            sh 'npm run lint'
+
+            echo "Run tests with coverage check"
+            sh 'npm run test:coverage'
+
+            echo "Run tests"
+            sh 'npm run test'
+            
+            echo "Run documentation linter"
+            sh 'npm run lint:docs'
+
+            echo "Verify if documentation is up-to-date"
+            sh 'npm run test:docs'
+          }
+        }
+      }
+    }
+    stage('Deploy') {
+      when {
+        expression { return branch =~ /^publish-v.*/ }
+      }
+      steps {
+        container('node'){
+          dir("${env.WORKSPACE}/${env.PROJECT}") {
+            withVault([configuration: configuration, vaultSecrets: secrets]) {
+              echo "Authenticate with registry"
+              sh "echo '//registry.npmjs.org/:_authToken=${env.NPM_TOKEN}' > .npmrc"
+              
+              echo "Publish package"
+              sh 'npm publish'
+
+              echo "Setup git user"
+              sh '''
+                git config user.email "cosmo-admin@github.com"
+                git config user.name "cosmo-admin-CircleCI"
```

Comment:
Maybe to be aligned across different projects and avoid potential inconsistency you can also put it to some Jenkins/Vault configuration and we will just get the values from there in `cloudify-ui-common`, `cloudify-ui-components` and `cloudify-blueprint-topology`. Wdyt?

Reply:
I will put it in the shared library in the future when all workflows will be migrated I will know what exactly to extract